opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetrySpanBuilderImpl.java

```diff
@@ -0,0 +1,229 @@
+  private final List<io.opentelemetry.api.trace.SpanContext> otelParentLinks = new ArrayList<>();
+  @Nullable private final Span ocParent;
+  @Nullable private final SpanContext ocRemoteParentSpanContext;
+  @Nullable private Sampler ocSampler;
+  @Nullable private Boolean recordEvents;
+  @Nullable private io.opentelemetry.api.trace.Span.Kind otelKind;
+
+  @Override
+  public SpanBuilder setSampler(Sampler sampler) {
+    this.ocSampler = checkNotNull(sampler, "sampler");
+    return this;
+  }
+
+  @Override
+  public SpanBuilder setParentLinks(List<Span> parentLinks) {
+    this.ocParentLinks = checkNotNull(parentLinks, "parentLinks");
+    for (Span parent : parentLinks) {
+      this.otelParentLinks.add(SpanConverter.mapSpanContext(parent.getContext()));
+    }
+    return this;
+  }
+
+  @Override
+  public SpanBuilder setRecordEvents(boolean recordEvents) {
+    this.recordEvents = recordEvents;
+    return this;
+  }
+
+  @Override
+  public SpanBuilder setSpanKind(@Nullable Kind kind) {
+    this.otelKind = mapKind(kind);
+    return this;
+  }
+
+  @Override
+  public Span startSpan() {
+    // To determine whether to sample this span
+    TraceParams ocActiveTraceParams = options.traceConfig.getActiveTraceParams();
+    Random random = options.randomHandler.current();
+    TraceId ocTraceId;
+    SpanId ocSpanId = SpanId.generateRandomId(random);
+    Tracestate ocTracestate = OC_TRACESTATE_DEFAULT;
+    SpanContext ocParentContext = null;
+    Boolean hasRemoteParent = null;
+    if (ocRemoteParentSpanContext != null && ocRemoteParentSpanContext.isValid()) {
+      ocParentContext = ocRemoteParentSpanContext;
+      hasRemoteParent = Boolean.TRUE;
+      ocTraceId = ocParentContext.getTraceId();
+      ocTracestate = ocParentContext.getTracestate();
+    } else if (ocParent != null && ocParent.getContext().isValid()) {
+      ocParentContext = ocParent.getContext();
+      hasRemoteParent = Boolean.FALSE;
+      ocTraceId = ocParentContext.getTraceId();
+      ocTracestate = ocParentContext.getTracestate();
+    } else {
+      // New root span.
+      ocTraceId = TraceId.generateRandomId(random);
+    }
+    TraceOptions ocTraceOptions =
+        makeSamplingDecision(
+                ocParentContext,
+                hasRemoteParent,
+                name,
+                ocSampler,
+                ocParentLinks,
+                ocTraceId,
+                ocSpanId,
+                ocActiveTraceParams)
+            ? OC_SAMPLED_TRACE_OPTIONS
+            : OC_NOT_SAMPLED_TRACE_OPTIONS;
+    if (!ocTraceOptions.isSampled() && !Boolean.TRUE.equals(recordEvents)) {
+      return OpenTelemetryNoRecordEventsSpanImpl.create(
+          SpanContext.create(ocTraceId, ocSpanId, ocTraceOptions, ocTracestate));
+    }
+
+    // If sampled
+    io.opentelemetry.api.trace.SpanBuilder otelSpanBuilder =
+        OTEL_TRACER
+            .spanBuilder(name)
+            .setStartTimestamp(options.clock.nowNanos(), TimeUnit.NANOSECONDS);
```

Comment:
Do you think we need to support the OC clock? I'd recommend removing this line if it seems reasonable.

Reply:
The clock is created at https://github.com/zoercai/opentelemetry-java/blob/4e7d8b25cb24466f5a678cfd67dd5769fe599d4e/opencensus-shim/src/main/java/io/opentelemetry/opencensusshim/OpenTelemetryTraceComponentImpl.java#L33
and is publicly available via `Tracing.java`, so it seems like I should use it here just in case. Let me know if you think it's unnecessary though!

Reply:
Customizing the OC clock seems like a very small use case - it seems unlikely to do any better than what OTel will provide (e.g., we updated the Clock to use Java 9+ for improved precision) so it could be considered a good thing to not respect the OC clock. But no personal opinion food for thought on what could be better here.

Reply:
I've removed the clock